src/rgw/rgw_kmip_client.h

```diff
@@ -0,0 +1,67 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#ifndef CEPH_RGW_KMIP_CLIENT_H
+#define CEPH_RGW_KMIP_CLIENT_H
+
+class RGWKMIPManager;
+
+class RGWKMIPTransceiver {
+public:
+	enum kmip_operation {
+		CREATE,
+		LOCATE,
+		GET,
+		GET_ATTRIBUTES,
+		GET_ATTRIBUTE_LIST,
+		DESTROY
+	};
+	CephContext *cct;
+	kmip_operation operation;
+	char *name = 0;
+	char *unique_id = 0;
+	// output - must free
+	char *out = 0;		// unique_id, several
+	struct {		// unique_ids, locate
+		char **strings;
+		int string_count;
+	} outlist[1] = {{0, 0}};
+	struct {		// key, get
+		unsigned char *data;
+		int keylen;
+	} outkey[1] = {0, 0};
+	// end must free
+	int ret;
+	bool done;
+	ceph::mutex lock = ceph::make_mutex("rgw_kmip_req::lock");
+	ceph::condition_variable cond;
+
+	int wait(optional_yield y);
```

Comment:
if `wait()` and `process()` don't support yielding on the coroutine, they shouldn't take an `optional_yield`